licenseApp.js

```diff
@@ -1,22 +1,31 @@
-var express = require('express');
-var systemd = require('systemd');
-var autoquit = require('autoquit');
-var http = require('http');
 
-var crawler = require('./licenseCrawler');
+const GLib = imports.gi.GLib;
+const Soup = imports.gi.Soup;
+const licenseCrawler = imports.licenseCrawler;
 
-var app = express();
-var port = 'systemd';
-if ('development' == app.get('env')) {
-    port = 3010;
+const DEFAULT_PORT = 3010;
+
+function serverListen(server) {
+    // If we're being started by systemd, listen on the 3rd fd.
+    if (GLib.getenv('LISTEN_FDS') !== null) {
+        server.listen_fd(3, 0);
+    } else {
+        server.listen_local(DEFAULT_PORT, 0);
+    }
 }
 
-app.set('port', process.env.PORT || port);
-app.get('/', crawler.getLicenseList);
+let mainloop = new GLib.MainLoop(null, false);
+
+let server = new Soup.Server();
+server.add_handler('/', function(server, msg, path, query, client) {
+    let res = {
+        write: function(text) { return msg.response_body.append(text); },
+    };
+
+    msg.status_code = 200;
+    msg.response_headers.set_content_type('text/html', { charset: 'UTF-8' });
 
-var server = http.createServer(app);
-server.autoQuit({ timeOut: 120 });
-server.listen(app.get('port'));
-server.on('listening', function() {
-    console.log('Express server listening on port ' + app.get('port'));
+    licenseCrawler.getLicenseList(res);
```

Comment:
https://github.com/GNOME/gjs/blob/master/examples/http-server.js doesn't -- should I?